### PR TITLE
Keep remote pipelines up to date documentation

### PR DESCRIPTION
**Pull request by fabriziosta**

### CODE REVIEW

File: docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc [hunk 1]
```diff
@@ -0,0 +1,61 @@
+= Maintaining remote Tekton pipelines up to date
+
+== The problem
+
+Although Tekton remote pipelines is a huge improvement for having a centralized and shared repo for Pipelines and Tasks, it comes with a drawback; the Pipeline CRDs will not be stored in the Konflux Component repository anymore, so MintMaker will not be able to update the TaskRef digests for us.
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc:5]: I feel like this problem is slightly incomplete. Is the issue that you are trying to describe here one where a common pipeline is self-managed based on Konflux tasks? Are you creating a self-defined "component template" which is then used as the source for the pipeline definition for other components?

If so, how should the other components reference the pipelines and tasks that are built here?

File: docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+.
+├── .tekton
+│   ├── <repository-name>-pipelines-pull-request.yaml # Used by Konflux to listen to pull requests
+│   └── <repository-name>-pipelines-push.yaml # Used by Konflux to listen to pushes
+├── pipelines
+│   ├── remote-pipeline1.yaml
+│   └── remote-pipeline2.yaml
+├── tasks
+│   ├── remote-task1.yaml
+│   └── remote-task2.yaml
```
Comment (arewm) [inline docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc:35]: What is the difference between all of these files? What should go into each of these?

**Review by arewm — COMMENTED** [review]
Some questions asked to start discussions. This topic also seems like it might fit better within the "workflow" section similar to the git submodule approach.

File: docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (fabriziosta) [reply docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc:35]: To highlight the difference between shared pipelines and non-shared. 
The  .tekton pipelines the default ones, and they are used by Konflux. The tekton pipelines inside the "pipelines" folder are the shared ones; those are the pipelines that will be referenced from the PipelineRuns.

Comment (fabriziosta) [reply docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc:35]: The several remote pipelines that could be run from different components. I.E.
- multi aarch pipeline
- hermetic pipeline
- pipeline with disabled checks or a very few tasks in it
- pipeline with pre-fetch enabled
- custom pipeline with untrusted Tasks
and so on.

File: docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc (hunk 1 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc:5]: mmh good point, I don't have a strong opinion on where this should be placed, I'm okay with that. Should I also emphasize the advantages of using remote pipelines? Like three different sections: the pros (and how to use remote pipelines), the drawbacks, the solution to the drawback.

Comment (arewm) [reply docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc:5]: Yes, I think that sounds great! You can use the existing workflow as a reference for some of the content that we included before. We can also generalize the workflows section (i.e. title) as it is a little overly specific to the git submodule approach.

Comment (fabriziosta) [reply docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc]: Moved to workflows,added advantages section and ready for the next round of reviews :)

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
+= Maintaining remote Tekton pipelines up to date
+
+[IMPORTANT]
+====
+* This article took for granted the reader knows what a link:https://tekton.dev/docs/pipelines/resolution/[Tekton remote pipeline] is. If this is not the case, please read more about this feature before proceeding.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:5]:
```suggestion
* This article assumed knowledge of link:https://tekton.dev/docs/pipelines/resolution/[Tekton remote pipelines]. If this is not the case, please read more about this feature before proceeding.
```

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+====
+
+== The advantage of using a remote pipeline
+
+This approach allows a team or an individual contributor to manage plenty of Pipelines from a single repository as source of truth. But there also other good reasons:
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:10]:
```suggestion
This approach allows a team or an individual contributor to manage multiple custom Pipelines from a single repository as source of truth. But there also other good reasons:
```

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+. Pipelines are stored in a Git repository and fetched during execution, eliminating the need to maintain local or static pipeline definitions.
+. Teams can leverage Git’s versioning to access specific pipeline branches
+. In case of issues, it's very simple to rollback to a previous pipeline version, and apply it globally among the Konflux Components.
+. Reduced maintenance overhead
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:15]:
```suggestion
. Reduced maintenance overhead of keeping task references up to date across multiple components.
```

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+* This article took for granted the reader knows what a link:https://tekton.dev/docs/pipelines/resolution/[Tekton remote pipeline] is. If this is not the case, please read more about this feature before proceeding.
+====
+
+== The advantage of using a remote pipeline
+
+This approach allows a team or an individual contributor to manage plenty of Pipelines from a single repository as source of truth. But there also other good reasons:
+
+. Pipelines are stored in a Git repository and fetched during execution, eliminating the need to maintain local or static pipeline definitions.
+. Teams can leverage Git’s versioning to access specific pipeline branches
+. In case of issues, it's very simple to rollback to a previous pipeline version, and apply it globally among the Konflux Components.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:14]: It is possible to roll back to previous versions without this remote pipeline strategy as well. I guess that the suggestion is to rollback the reference pipeline commit based on the use of a floating branch tag? If you are suggesting that it is easy to roll back, I think we should document that in this page.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+====
+* This article took for granted the reader knows what a link:https://tekton.dev/docs/pipelines/resolution/[Tekton remote pipeline] is. If this is not the case, please read more about this feature before proceeding.
+====
+
+== The advantage of using a remote pipeline
+
+This approach allows a team or an individual contributor to manage plenty of Pipelines from a single repository as source of truth. But there also other good reasons:
+
+. Pipelines are stored in a Git repository and fetched during execution, eliminating the need to maintain local or static pipeline definitions.
+. Teams can leverage Git’s versioning to access specific pipeline branches
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:13]:
```suggestion
. Teams can leverage Git’s versioning to access specific pipeline branches.
```

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    resolver: git
+    params:
+      - name: org
+        value: redhat
+      - name: scmType
+        value: gitlab
+      - name: serverURL
+        value: https://gitlab.cee.redhat.com
+      - name: url
+        value: "https://gitlab.cee.redhat.com/your-group-name/your-repo-name.git"
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:36]: Let's use an externally familiar URL here.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+      - name: revision
+        value: main # Specify a branch to target
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:38]: It is possible to reference a specific revision as well. I guess the benefit of using a "floating" reference like this is that the rest of the components get updates without having to get a nudge. They will get the updates in their next commit.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+To start referencing a remote Pipeline from a PipelineRun using a *git resolver*, replace *pipelineSpec* with *pipelineRef*:
+
+[source,yaml]
+----
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+metadata: ... # Omitted for brevity
+spec:
+  params: ... # Omitted for brevity
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:25]: Instead of adding comments into the yaml blocks, we can use the `<.>` pattern for adding footnotes like we do elsewhere (i.e. [create-release.adoc](https://github.com/konflux-ci/docs/blob/main/docs/modules/ROOT/pages/advanced-how-tos/releasing/create-release.adoc?plain=1)).

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+      - name: url
+        value: "https://gitlab.cee.redhat.com/your-group-name/your-repo-name.git"
+      - name: revision
+        value: main # Specify a branch to target
+      - name: pathInRepo
+        value: "pipelines/your-pipeline-name.yaml" # Internal repository path to your pipeline definition YAML file
+      - name: token
+        value: pipelines-as-code-secret # Name of the Secret
+      - name: tokenKey
+        value: password # Name of the attribute within your Secret, that stores the password/token
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:44]: Does this refer to a specific secret that we mention elsewhere in the documentation? If so, can we refer to the page that describes the process as well?

Is this secret ever _not_ needed? I guess if the target can be anonymously accessed (like in a public GitHub repo)?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+  workspaces: ... # Omitted for brevity
+----
+
+== The drawback of having a remote pipeline
+
+Although Tekton remote pipelines is a huge improvement, it comes with a drawback; the Pipeline CRDs will not be stored in the Konflux Component repository anymore, so MintMaker will not be able to update the TaskRef digests for us.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:50]: I think that this slightly misrepresents the situation. There is no pipeline CRD, but the Component CR has a reference to the specific pipeline that was used/requested when onboarding (or [reconfigured](https://github.com/konflux-ci/docs/blob/main/docs/modules/ROOT/pages/how-tos/configuring/reconfiguring-build-pipeline.adoc)).

MintMaker doesn't update the TaskRefs because anything is in the CR, but because those task references are in a "known" location.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+== The solution
+
+To workaround the above issue, the suggested approach is to onboard the repository where the shared pipelines are stored as a Konflux Component, so that Konflux and MintMaker will be able to discover and update digests for us.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:54]: Should we give this location of the shared pipelines a name to make it easier to refer to? For example a common pipeline component or reference component (I don't think that my naming suggestions are that good right now).

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+.*Prerequisites*
+
+* Having a repository where to store shared Pipelines and Tasks.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:58]: Is this not a prerequisite for the entire workflow? Or is it possible to achieve this workflow without this?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 15]
```diff
@@ -0,0 +1,107 @@
+= Maintaining remote Tekton pipelines up to date
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:1]: General comment on structure of this page. If we create a structure similar to the git submodule approach, I think that this page would have sections like:

(brief description of the problem we are trying to solve)
* Implementing this guideline
  * Basic definitions
  * Create a reference component (see comment about naming the component)
  * Configure renovate
  * Configure additional components with the remote resolver
* Benefits of this guideline
* Potential drawbacks

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
++
+[source,shell]
+----
+.
+├── .tekton
+│   ├── <repository-name>-pipelines-pull-request.yaml # Used by Konflux to listen to pull requests
+│   └── <repository-name>-pipelines-push.yaml # Used by Konflux to listen to pushes
+├── pipelines
+│   ├── remote-pipeline1.yaml
+│   └── remote-pipeline2.yaml
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:78]: I feel like the general recommendation _might_ be to have a reference component build for each of these pipelines (i.e. if you have a single arch and multi-arch pipeline, one component that exercises each in a way that is representative for your other components). This would enable the CI for _this_ repo to catch when task/pipeline updates will present issues downstream.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+. Add Tekton Pipelines and Tasks in the repository. A suggested repository structure would be:
+
++
+[source,shell]
+----
+.
+├── .tekton
+│   ├── <repository-name>-pipelines-pull-request.yaml # Used by Konflux to listen to pull requests
+│   └── <repository-name>-pipelines-push.yaml # Used by Konflux to listen to pushes
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:75]: These PipelineRun files should reference the common pipelines (and by extension common tasks) below, right?

**Review by arewm — COMMENTED** [review]
I like how this page is shaping up!

Instead of using `Konflux` directly, we use `{ProductName}`.

File: docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ernesgonzalez33) [reply docs/modules/ROOT/pages/advanced-how-tos/keep-remote-pipelines-up-to-date.adoc]: I'm with Fabrizio on this one! `.tekton` in the end is what PaC uses for the pipelines it manages. That doesn't mean you can't do anything you want on that dir, but I prefer keeping it differentiated

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 5 above)
Comment (ernesgonzalez33) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:15]: There is more maintenance overhead. For example: if you need custom tasks in all of your pipelines, if you manage different versions through the project controller you need to update all branches as well if you change something in a task

Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc]: Mmmh, why would you target a different revision/branch everywhere? Can't you just test your changes on a branch updating a single component and then open a PR against main/devel branch?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 6 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:14]: > I guess that the suggestion is to rollback the reference pipeline commit based on the use of a floating branch tag?
Yup, that's exactly what I meant. Let me specify it better

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 13 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:54]: Could you rephrase your idea? I am not sure I get it

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 14 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:58]: Yeah it's a prerequisite for the entire workflow, should I move it under another section or paragraph?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 16 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:78]: Are you suggesting to rename the pipelines with some more descriptive namings?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+
+Although Tekton remote pipelines is a huge improvement, it comes with a drawback; the Pipeline CRDs will not be stored in the Konflux Component repository anymore, so MintMaker will not be able to update the TaskRef digests for us.
+
+== The solution
+
+To workaround the above issue, the suggested approach is to onboard the repository where the shared pipelines are stored as a Konflux Component, so that Konflux and MintMaker will be able to discover and update digests for us.
+
+.*Prerequisites*
+
+* Having a repository where to store shared Pipelines and Tasks.
```
Comment (swadeley) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:58]:
```suggestion
* Having a repository to store shared Pipelines and Tasks.
```

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+}
+----
+
++
+[NOTE]
+====
+* MintMaker will read the default configuration from "https://docs.renovatebot.com/renovate-schema.json" and will extend it using the subsequent custom settings.
+====
+
+. xref:how-tos/creating.adoc[Create a Konflux Application and a Konflux Component] where to reference the repository and branch name.
```
Comment (swadeley) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:107]:
```suggestion
. xref:how-tos/creating.adoc[Create a Konflux Application and a Konflux Component] to reference the repository and branch name.
```

File: docs/modules/ROOT/pages/how-tos/_nav.adoc [hunk 20]
```diff
@@ -35,3 +35,4 @@
 ** xref:how-tos/deleting.adoc[Deleting applications and components]
 ** xref:how-tos/workflows/index.adoc[Guidelines for {ProductName} integrated repositories]
 *** xref:how-tos/workflows/git-submodules.adoc[Building upstream projects with git submodules]
+*** xref:how-tos/workflows/keep-remote-pipelines-up-to-date.adoc[Maintaining remote pipelines up to date]
```
Comment (swadeley) [inline docs/modules/ROOT/pages/how-tos/_nav.adoc:38]:
```suggestion
*** xref:how-tos/workflows/keep-remote-pipelines-up-to-date.adoc[Maintaining remote pipelines]
```
I think "Maintaining" implies "keeping up to date"

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 15 above)
Comment (swadeley) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:1]:
```suggestion
= Maintaining remote Tekton pipelines
```
reason as above

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 21]
```diff
@@ -0,0 +1,107 @@
+= Maintaining remote Tekton pipelines up to date
+
+[IMPORTANT]
+====
+* This article assumed knowledge of link:https://tekton.dev/docs/pipelines/resolution/[Tekton remote pipelines]. If this is not the case, please read more about this feature before proceeding.
```
Comment (swadeley) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:5]:
```suggestion
* This article assumes knowledge of link:https://tekton.dev/docs/pipelines/resolution/[Tekton remote pipelines]. If this is not the case, please read more about this feature before proceeding.
```

**Comment by fabriziosta** [comment]
> I like how this page is shaping up!
> 
> Instead of using `Konflux` directly, we use `{ProductName}`.

Done!

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 6 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:14]: I was not thinking about using a floating branch tag, I have not tested it despite I suppose it works out of the box. I would leave it as it is and we could always edit this page in the future eventually to add more details or strategies.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 15 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc]: This change required a hard refactor of the documentation. Please re-review it.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 22]
```diff
@@ -0,0 +1,113 @@
...
+
+== Implementing this guideline
+
+.*Prerequisites*
+
+* Having a repository to store shared Pipelines and Tasks.
+
+[IMPORTANT]
+====
+If the repository is private, be sure to have a Secret in place on {ProductName} to be able to clone it
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:16]:
```suggestion
If the repository is private, be sure to have a Secret in your {ProductName} workspace to be able to clone it
```
Is this the same as the secret configuration for cloning repositories? Is there any additional documentation that we can link to in order to help manage the step?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc [hunk 23]
```diff
@@ -0,0 +1,113 @@
...
+
+. Add Tekton Pipelines and Tasks in the repository. A suggested repository structure would be:
+
++
+[source,shell]
+----
+.
+├── .tekton
+│   ├── <repository-name>-pipelines-pull-request.yaml # Used by {ProductName} to listen to pull requests
+│   └── <repository-name>-pipelines-push.yaml # Used by {ProductName} to listen to pushes
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:31]: nit: I don't know if listen is the right action. Maybe trigger pipelines on those events?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 22 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:16]: It's the one specified in the pipelineRef, in the example is named "pipelines-as-code-secret" 
```
pipelineRef:
    resolver: git
    params:
      - name: org
        value: redhat
      - name: scmType
        value: gitlab
      - name: serverURL
        value: https://gitlab.cee.redhat.com
      - name: url
        value: "https://gitlab.cee.redhat.com/your-group-name/your-repo-name.git"
      - name: revision
        value: main # Specify a branch to target
      - name: pathInRepo
        value: "pipelines/your-pipeline-name.yaml" # Internal repository path to your pipeline definition YAML file
      - name: token
        value: pipelines-as-code-secret # Name of the Secret
      - name: tokenKey
        value: password # Name of the attribute within your Secret, that stores the password/token
```

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 23 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:31]: What about "Use by Konflux to react to pull requests/pushes"?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 9 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:65]: I wanted to show that's possible to set a branch name. Should we replace it with "{{revision}}"?

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 11 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:71]: - It's mentioned https://konflux.pages.redhat.com/docs/users/how-tos/configuring/creating-secrets.html
- For the latter question, I am not sure this is mandatory to be specified, but I wanted to show the more complex example, so that the easier one (without secret) would be simpler to reproduce.

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 23 above)
Comment (swadeley) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc:31]: # Used by {ProductName} to monitor pull requests
# Used by {ProductName} to monitor pushes

Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc]: Accepted swadeley's suggestion. Ty!

File: docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc (hunk 22 above)
Comment (fabriziosta) [reply docs/modules/ROOT/pages/how-tos/workflows/keep-remote-pipelines-up-to-date.adoc]: Accepted your suggestion and added a reference to this page: https://konflux.pages.redhat.com/docs/users/how-tos/configuring/creating-secrets.html

**Review by arewm — APPROVED** [review]
Thanks for this!
